{"cells":[{"cell_type":"markdown","id":"cd60e6bf","metadata":{"id":"cd60e6bf"},"source":["# 🔬Differentially Expressed Genes (DEG) Selection: Analyzing RNASeq data with DESeq2 package\n","\n","This notebook finds the top differentially expressed genes (DEGs) using:\n","- ✅ DESeq2 with paired design in R\n","\n","# Input files expected:\n","- `long_format_tumor_normal_samples.csv`: Sample metadata (Columns: SampleID, condition [Tumor/Normal], PatientID)\n","- `tcga_luad_raw_expression_data.csv`: RNA-seq matrix - raw fold-change values (rows = sample IDs, columns = genes). Downloaded from UCSC Xena public links for TCGA-LUAD: https://tcga.xenahubs.net/download/TCGA.LUAD.sampleMap/HiSeqV2.gz\n","- `expression_data_matched.csv`: RNA-seq data for samples which have matching methylation data (rows = sample IDs, columns = genes)\n","\n","# Output produced:\n","- `DESeq2_paired_DEG_results.csv`: DESeq2 output (diffrencialy expressed gene (Columns: `baseMean`, `log2FoldChange`, `lfcSE`, `pvalue`, `padj`\n","\n","\n"]},{"cell_type":"markdown","source":["##1) Setup & upload\n","###1.1) Getting the list of tumor/normal sampleId with both expression and methylation data"],"metadata":{"id":"EB9qo5K-7EZL"},"id":"EB9qo5K-7EZL"},{"cell_type":"code","source":["# If running in Colab, use this to upload files. Otherwise ensure files exist in working directory.\n","try:\n","    from google.colab import files\n","    IN_COLAB = True\n","except Exception:\n","    IN_COLAB = False\n","print('IN_COLAB =', IN_COLAB)\n","\n","if IN_COLAB:\n","    print('Please upload long_format_tumor_normal_samples.csv, expression_data_matched.csv')\n","    uploaded = files.upload()\n","    print('Uploaded:', list(uploaded.keys()))"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":163},"id":"zMRyw5wGe-i5","executionInfo":{"status":"ok","timestamp":1761095880976,"user_tz":420,"elapsed":109486,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}},"outputId":"92212832-4cb1-48db-aa2e-05ad0c029338"},"id":"zMRyw5wGe-i5","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["IN_COLAB = True\n","Please upload long_format_tumor_normal_samples.csv, expression_data_matched.csv\n"]},{"output_type":"display_data","data":{"text/plain":["<IPython.core.display.HTML object>"],"text/html":["\n","     <input type=\"file\" id=\"files-56762b99-9fc5-488d-83c5-5b351e5bfaba\" name=\"files[]\" multiple disabled\n","        style=\"border:none\" />\n","     <output id=\"result-56762b99-9fc5-488d-83c5-5b351e5bfaba\">\n","      Upload widget is only available when the cell has been executed in the\n","      current browser session. Please rerun this cell to enable.\n","      </output>\n","      <script>// Copyright 2017 Google LLC\n","//\n","// Licensed under the Apache License, Version 2.0 (the \"License\");\n","// you may not use this file except in compliance with the License.\n","// You may obtain a copy of the License at\n","//\n","//      http://www.apache.org/licenses/LICENSE-2.0\n","//\n","// Unless required by applicable law or agreed to in writing, software\n","// distributed under the License is distributed on an \"AS IS\" BASIS,\n","// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n","// See the License for the specific language governing permissions and\n","// limitations under the License.\n","\n","/**\n"," * @fileoverview Helpers for google.colab Python module.\n"," */\n","(function(scope) {\n","function span(text, styleAttributes = {}) {\n","  const element = document.createElement('span');\n","  element.textContent = text;\n","  for (const key of Object.keys(styleAttributes)) {\n","    element.style[key] = styleAttributes[key];\n","  }\n","  return element;\n","}\n","\n","// Max number of bytes which will be uploaded at a time.\n","const MAX_PAYLOAD_SIZE = 100 * 1024;\n","\n","function _uploadFiles(inputId, outputId) {\n","  const steps = uploadFilesStep(inputId, outputId);\n","  const outputElement = document.getElementById(outputId);\n","  // Cache steps on the outputElement to make it available for the next call\n","  // to uploadFilesContinue from Python.\n","  outputElement.steps = steps;\n","\n","  return _uploadFilesContinue(outputId);\n","}\n","\n","// This is roughly an async generator (not supported in the browser yet),\n","// where there are multiple asynchronous steps and the Python side is going\n","// to poll for completion of each step.\n","// This uses a Promise to block the python side on completion of each step,\n","// then passes the result of the previous step as the input to the next step.\n","function _uploadFilesContinue(outputId) {\n","  const outputElement = document.getElementById(outputId);\n","  const steps = outputElement.steps;\n","\n","  const next = steps.next(outputElement.lastPromiseValue);\n","  return Promise.resolve(next.value.promise).then((value) => {\n","    // Cache the last promise value to make it available to the next\n","    // step of the generator.\n","    outputElement.lastPromiseValue = value;\n","    return next.value.response;\n","  });\n","}\n","\n","/**\n"," * Generator function which is called between each async step of the upload\n"," * process.\n"," * @param {string} inputId Element ID of the input file picker element.\n"," * @param {string} outputId Element ID of the output display.\n"," * @return {!Iterable<!Object>} Iterable of next steps.\n"," */\n","function* uploadFilesStep(inputId, outputId) {\n","  const inputElement = document.getElementById(inputId);\n","  inputElement.disabled = false;\n","\n","  const outputElement = document.getElementById(outputId);\n","  outputElement.innerHTML = '';\n","\n","  const pickedPromise = new Promise((resolve) => {\n","    inputElement.addEventListener('change', (e) => {\n","      resolve(e.target.files);\n","    });\n","  });\n","\n","  const cancel = document.createElement('button');\n","  inputElement.parentElement.appendChild(cancel);\n","  cancel.textContent = 'Cancel upload';\n","  const cancelPromise = new Promise((resolve) => {\n","    cancel.onclick = () => {\n","      resolve(null);\n","    };\n","  });\n","\n","  // Wait for the user to pick the files.\n","  const files = yield {\n","    promise: Promise.race([pickedPromise, cancelPromise]),\n","    response: {\n","      action: 'starting',\n","    }\n","  };\n","\n","  cancel.remove();\n","\n","  // Disable the input element since further picks are not allowed.\n","  inputElement.disabled = true;\n","\n","  if (!files) {\n","    return {\n","      response: {\n","        action: 'complete',\n","      }\n","    };\n","  }\n","\n","  for (const file of files) {\n","    const li = document.createElement('li');\n","    li.append(span(file.name, {fontWeight: 'bold'}));\n","    li.append(span(\n","        `(${file.type || 'n/a'}) - ${file.size} bytes, ` +\n","        `last modified: ${\n","            file.lastModifiedDate ? file.lastModifiedDate.toLocaleDateString() :\n","                                    'n/a'} - `));\n","    const percent = span('0% done');\n","    li.appendChild(percent);\n","\n","    outputElement.appendChild(li);\n","\n","    const fileDataPromise = new Promise((resolve) => {\n","      const reader = new FileReader();\n","      reader.onload = (e) => {\n","        resolve(e.target.result);\n","      };\n","      reader.readAsArrayBuffer(file);\n","    });\n","    // Wait for the data to be ready.\n","    let fileData = yield {\n","      promise: fileDataPromise,\n","      response: {\n","        action: 'continue',\n","      }\n","    };\n","\n","    // Use a chunked sending to avoid message size limits. See b/62115660.\n","    let position = 0;\n","    do {\n","      const length = Math.min(fileData.byteLength - position, MAX_PAYLOAD_SIZE);\n","      const chunk = new Uint8Array(fileData, position, length);\n","      position += length;\n","\n","      const base64 = btoa(String.fromCharCode.apply(null, chunk));\n","      yield {\n","        response: {\n","          action: 'append',\n","          file: file.name,\n","          data: base64,\n","        },\n","      };\n","\n","      let percentDone = fileData.byteLength === 0 ?\n","          100 :\n","          Math.round((position / fileData.byteLength) * 100);\n","      percent.textContent = `${percentDone}% done`;\n","\n","    } while (position < fileData.byteLength);\n","  }\n","\n","  // All done.\n","  yield {\n","    response: {\n","      action: 'complete',\n","    }\n","  };\n","}\n","\n","scope.google = scope.google || {};\n","scope.google.colab = scope.google.colab || {};\n","scope.google.colab._files = {\n","  _uploadFiles,\n","  _uploadFilesContinue,\n","};\n","})(self);\n","</script> "]},"metadata":{}},{"output_type":"stream","name":"stdout","text":["Saving expression_data_matched.csv to expression_data_matched.csv\n","Saving long_format_tumor_normal_samples.csv to long_format_tumor_normal_samples.csv\n","Uploaded: ['expression_data_matched.csv', 'long_format_tumor_normal_samples.csv']\n"]}]},{"cell_type":"code","execution_count":null,"id":"ea48559e","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":242},"id":"ea48559e","outputId":"22bffed4-175d-4bfc-8503-e33fa40cb87e","executionInfo":{"status":"ok","timestamp":1761069451320,"user_tz":420,"elapsed":569,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}},"collapsed":true},"outputs":[{"output_type":"stream","name":"stdout","text":["tumor/normal sample paired count which have methylation data: 29\n","tumor/normal sample paired count with both methylation & expression data: 18\n"]},{"output_type":"execute_result","data":{"text/plain":["    SampleID_Tumor condition_Tumor     PatientID  SampleID_Normal  \\\n","0  TCGA-44-6778-01           Tumor  TCGA-44-6778  TCGA-44-6778-11   \n","1  TCGA-50-5931-01           Tumor  TCGA-50-5931  TCGA-50-5931-11   \n","3  TCGA-44-2668-01           Tumor  TCGA-44-2668  TCGA-44-2668-11   \n","4  TCGA-44-2665-01           Tumor  TCGA-44-2665  TCGA-44-2665-11   \n","5  TCGA-44-6148-01           Tumor  TCGA-44-6148  TCGA-44-6148-11   \n","\n","  condition_Normal  \n","0           Normal  \n","1           Normal  \n","3           Normal  \n","4           Normal  \n","5           Normal  "],"text/html":["\n","  <div id=\"df-d659f832-2947-4cf3-8dc2-550d5285c29f\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>SampleID_Tumor</th>\n","      <th>condition_Tumor</th>\n","      <th>PatientID</th>\n","      <th>SampleID_Normal</th>\n","      <th>condition_Normal</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>TCGA-44-6778-01</td>\n","      <td>Tumor</td>\n","      <td>TCGA-44-6778</td>\n","      <td>TCGA-44-6778-11</td>\n","      <td>Normal</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>TCGA-50-5931-01</td>\n","      <td>Tumor</td>\n","      <td>TCGA-50-5931</td>\n","      <td>TCGA-50-5931-11</td>\n","      <td>Normal</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>TCGA-44-2668-01</td>\n","      <td>Tumor</td>\n","      <td>TCGA-44-2668</td>\n","      <td>TCGA-44-2668-11</td>\n","      <td>Normal</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>TCGA-44-2665-01</td>\n","      <td>Tumor</td>\n","      <td>TCGA-44-2665</td>\n","      <td>TCGA-44-2665-11</td>\n","      <td>Normal</td>\n","    </tr>\n","    <tr>\n","      <th>5</th>\n","      <td>TCGA-44-6148-01</td>\n","      <td>Tumor</td>\n","      <td>TCGA-44-6148</td>\n","      <td>TCGA-44-6148-11</td>\n","      <td>Normal</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d659f832-2947-4cf3-8dc2-550d5285c29f')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-d659f832-2947-4cf3-8dc2-550d5285c29f button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-d659f832-2947-4cf3-8dc2-550d5285c29f');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-bcbf73cf-406a-4552-ab59-b6c7f9d31544\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-bcbf73cf-406a-4552-ab59-b6c7f9d31544')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-bcbf73cf-406a-4552-ab59-b6c7f9d31544 button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","variable_name":"valid_paired","summary":"{\n  \"name\": \"valid_paired\",\n  \"rows\": 18,\n  \"fields\": [\n    {\n      \"column\": \"SampleID_Tumor\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 18,\n        \"samples\": [\n          \"TCGA-44-6778-01\",\n          \"TCGA-50-5931-01\",\n          \"TCGA-49-6745-01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"condition_Tumor\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"Tumor\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PatientID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 18,\n        \"samples\": [\n          \"TCGA-44-6778\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"SampleID_Normal\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 18,\n        \"samples\": [\n          \"TCGA-44-6778-11\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"condition_Normal\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"Normal\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"}},"metadata":{},"execution_count":14}],"source":["import pandas as pd\n","import numpy as np\n","from scipy.stats import ttest_rel\n","from statsmodels.stats.multitest import fdrcorrection\n","\n","counts = pd.read_csv(\"expression_data_matched.csv\", index_col=0)\n","meta = pd.read_csv(\"long_format_tumor_normal_samples.csv\")\n","\n","# Match paired samples with methylation data\n","tumor_samples = meta[meta[\"condition\"] == \"Tumor\"]\n","normal_samples = meta[meta[\"condition\"] == \"Normal\"]\n","paired = pd.merge(tumor_samples, normal_samples, on=\"PatientID\", suffixes=(\"_Tumor\", \"_Normal\")).dropna()\n","print('tumor/normal sample paired count which have methylation data:', len(paired))\n","\n","\n","# Transpose count matrix: genes as index\n","counts_T = counts.T\n","\n","# Filter paired samples to keep only those present in counts_T columns\n","valid_tumor_samples = paired[paired[\"SampleID_Tumor\"].isin(counts_T.columns)]\n","valid_paired = valid_tumor_samples[valid_tumor_samples[\"SampleID_Normal\"].isin(counts_T.columns)]\n","print('tumor/normal sample paired count with both methylation & expression data:', len(valid_paired))\n","valid_paired.head()\n"]},{"cell_type":"markdown","source":["### 1.2) Installing DESeq2 library"],"metadata":{"id":"Ip1R-TAU7jT3"},"id":"Ip1R-TAU7jT3"},{"cell_type":"code","execution_count":null,"id":"35a97400","metadata":{"id":"35a97400","colab":{"base_uri":"https://localhost:8080/"},"outputId":"e9b23252-47fd-43e9-a512-a2528ec2197c","executionInfo":{"status":"ok","timestamp":1761001329920,"user_tz":420,"elapsed":3165,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}}},"outputs":[{"output_type":"stream","name":"stdout","text":["Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","r-base is already the newest version (4.5.1-1.2204.0).\n","0 upgraded, 0 newly installed, 0 to remove and 38 not upgraded.\n"]}],"source":["!apt-get install -y r-base\n"]},{"cell_type":"code","execution_count":null,"id":"86ab3922","metadata":{"id":"86ab3922","colab":{"base_uri":"https://localhost:8080/"},"outputId":"9fd7ede1-fcd5-48ac-c316-a3f971f286db","executionInfo":{"status":"ok","timestamp":1761001651147,"user_tz":420,"elapsed":316969,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}},"collapsed":true},"outputs":[{"output_type":"stream","name":"stdout","text":["\n","R version 4.5.1 (2025-06-13) -- \"Great Square Root\"\n","Copyright (C) 2025 The R Foundation for Statistical Computing\n","Platform: x86_64-pc-linux-gnu\n","\n","R is free software and comes with ABSOLUTELY NO WARRANTY.\n","You are welcome to redistribute it under certain conditions.\n","Type 'license()' or 'licence()' for distribution details.\n","\n","  Natural language support but running in an English locale\n","\n","R is a collaborative project with many contributors.\n","Type 'contributors()' for more information and\n","'citation()' on how to cite R or R packages in publications.\n","\n","Type 'demo()' for some demos, 'help()' for on-line help, or\n","'help.start()' for an HTML browser interface to help.\n","Type 'q()' to quit R.\n","\n","> \n","> install.packages(\"BiocManager\", repos=\"http://cran.us.r-project.org\")\n","> BiocManager::install(\"DESeq2\", update=FALSE, ask=FALSE)\n","Creating a new generic function for ‘aperm’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘append’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘as.data.frame’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘cbind’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘rbind’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘do.call’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘duplicated’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘anyDuplicated’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘eval’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘pmax’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘pmin’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘pmax.int’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘pmin.int’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘Reduce’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘Filter’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘Find’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘Map’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘Position’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘get’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘mget’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘grep’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘grepl’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘is.unsorted’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘lapply’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘sapply’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘mapply’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘match’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘order’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘paste’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘rank’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘rownames’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘colnames’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘saveRDS’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘table’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘tapply’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘unique’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘unsplit’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘which.min’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘which.max’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘IQR’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘mad’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘var’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘sd’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘xtabs’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘basename’ in package ‘BiocGenerics’\n","Creating a new generic function for ‘dirname’ in package ‘BiocGenerics’\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 000.init.c -o 000.init.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c allocMatrix2.c -o allocMatrix2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c anyMissing.c -o anyMissing.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c binCounts.c -o binCounts.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c binMeans.c -o binMeans.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c colCounts.c -o colCounts.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c colOrderStats.c -o colOrderStats.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c colRanges.c -o colRanges.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c diff2.c -o diff2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c indexByRow.c -o indexByRow.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c logSumExp.c -o logSumExp.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c mean2.c -o mean2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c naming.c -o naming.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c productExpSumLog.c -o productExpSumLog.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c psortKM.c -o psortKM.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowCounts.c -o rowCounts.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowCummaxs.c -o rowCummaxs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowCummins.c -o rowCummins.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowCumprods.c -o rowCumprods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowCumsums.c -o rowCumsums.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowDiffs.c -o rowDiffs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowLogSumExp.c -o rowLogSumExp.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowMads.c -o rowMads.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowMeans2.c -o rowMeans2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowMedians.c -o rowMedians.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowOrderStats.c -o rowOrderStats.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowRanges.c -o rowRanges.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowRanksWithTies.c -o rowRanksWithTies.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowSums2.c -o rowSums2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowVars.c -o rowVars.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c signTabulate.c -o signTabulate.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c sum2.c -o sum2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c validateIndices.c -o validateIndices.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c weightedMean.c -o weightedMean.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c weightedMedian.c -o weightedMedian.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c x_OP_y.c -o x_OP_y.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o matrixStats.so 000.init.o allocMatrix2.o anyMissing.o binCounts.o binMeans.o colCounts.o colOrderStats.o colRanges.o diff2.o indexByRow.o logSumExp.o mean2.o naming.o productExpSumLog.o psortKM.o rowCounts.o rowCummaxs.o rowCummins.o rowCumprods.o rowCumsums.o rowDiffs.o rowLogSumExp.o rowMads.o rowMeans2.o rowMedians.o rowOrderStats.o rowRanges.o rowRanksWithTies.o rowSums2.o rowVars.o signTabulate.o sum2.o validateIndices.o weightedMean.o weightedMedian.o x_OP_y.o -L/usr/lib/R/lib -lR\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S_enter.c -o S_enter.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c band.c -o band.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dbinom.c -o dbinom.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dens_haz.c -o dens_haz.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dens_int.c -o dens_int.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dens_odi.c -o dens_odi.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c density.c -o density.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ev_atree.c -o ev_atree.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ev_interp.c -o ev_interp.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ev_kdtre.c -o ev_kdtre.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ev_main.c -o ev_main.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ev_sphere.c -o ev_sphere.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ev_trian.c -o ev_trian.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c family.c -o family.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fitted.c -o fitted.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frend.c -o frend.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_adap.c -o lf_adap.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_dercor.c -o lf_dercor.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_fitfun.c -o lf_fitfun.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_nbhd.c -o lf_nbhd.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_robust.c -o lf_robust.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_vari.c -o lf_vari.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lf_wdiag.c -o lf_wdiag.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c lfstr.c -o lfstr.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c locfit.c -o locfit.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_chol.c -o m_chol.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_eigen.c -o m_eigen.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_icirc.c -o m_icirc.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_imont.c -o m_imont.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_isimp.c -o m_isimp.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_isphr.c -o m_isphr.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_jacob.c -o m_jacob.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_max.c -o m_max.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_qr.c -o m_qr.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_solve.c -o m_solve.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_svd.c -o m_svd.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c m_vector.c -o m_vector.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c math.c -o math.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c minmax.c -o minmax.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c pcomp.c -o pcomp.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c preplot.c -o preplot.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c prob.c -o prob.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c procv.c -o procv.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c scb.c -o scb.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c scb_cons.c -o scb_cons.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c scb_crit.c -o scb_crit.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c scb_iface.c -o scb_iface.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c simul.c -o simul.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c smisc.c -o smisc.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c startlf.c -o startlf.o\n","gcc -std=gnu17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c weight.c -o weight.o\n","gcc -std=gnu17 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o locfit.so S_enter.o band.o dbinom.o dens_haz.o dens_int.o dens_odi.o density.o ev_atree.o ev_interp.o ev_kdtre.o ev_main.o ev_sphere.o ev_trian.o family.o fitted.o frend.o lf_adap.o lf_dercor.o lf_fitfun.o lf_nbhd.o lf_robust.o lf_vari.o lf_wdiag.o lfstr.o locfit.o m_chol.o m_eigen.o m_icirc.o m_imont.o m_isimp.o m_isphr.o m_jacob.o m_max.o m_qr.o m_solve.o m_svd.o m_vector.o math.o minmax.o pcomp.o preplot.o prob.o procv.o scb.o scb_cons.o scb_crit.o scb_iface.o simul.o smisc.o startlf.o weight.o -L/usr/lib/R/lib -lR\n","checking whether the C++ compiler works... yes\n","checking for C++ compiler default output file name... a.out\n","checking for suffix of executables... \n","checking whether we are cross compiling... no\n","checking for suffix of object files... o\n","checking whether the compiler supports GNU C++... yes\n","checking whether g++ -std=gnu++17 accepts -g... yes\n","checking for g++ -std=gnu++17 option to enable C++11 features... none needed\n","checking how to run the C++ preprocessor... g++ -std=gnu++17 -E\n","checking whether the compiler supports GNU C++... (cached) yes\n","checking whether g++ -std=gnu++17 accepts -g... (cached) yes\n","checking for g++ -std=gnu++17 option to enable C++11 features... (cached) none needed\n","checking whether we have a suitable tempdir... /tmp\n","checking whether R CMD SHLIB can already compile programs using OpenMP... yes\n","checking LAPACK_LIBS... system LAPACK found\n","configure: creating ./config.status\n","config.status: creating inst/include/RcppArmadillo/config/RcppArmadilloConfigGenerated.h\n","config.status: creating src/Makevars\n","g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I../inst/include -DARMA_USE_CURRENT -I'/usr/lib/R/site-library/Rcpp/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c RcppArmadillo.cpp -o RcppArmadillo.o\n","g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I../inst/include -DARMA_USE_CURRENT -I'/usr/lib/R/site-library/Rcpp/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c RcppExports.cpp -o RcppExports.o\n","g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG -I../inst/include -DARMA_USE_CURRENT -I'/usr/lib/R/site-library/Rcpp/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c fastLm.cpp -o fastLm.o\n","g++ -std=gnu++17 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o RcppArmadillo.so RcppArmadillo.o RcppExports.o fastLm.o -fopenmp -llapack -lblas -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c AEbufs.c -o AEbufs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c DataFrame_class.c -o DataFrame_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Hits_class.c -o Hits_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c LLint_class.c -o LLint_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c List_class.c -o List_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_S4Vectors.c -o R_init_S4Vectors.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Rle_class.c -o Rle_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Rle_utils.c -o Rle_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SEXP_utils.c -o SEXP_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SimpleList_class.c -o SimpleList_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c anyMissing.c -o anyMissing.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c character_utils.c -o character_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c eval_utils.c -o eval_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c hash_utils.c -o hash_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c integer_utils.c -o integer_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c logical_utils.c -o logical_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c map_ranges_to_runs.c -o map_ranges_to_runs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c raw_utils.c -o raw_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c safe_arithm.c -o safe_arithm.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c sort_utils.c -o sort_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c subsetting_utils.c -o subsetting_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c vector_utils.c -o vector_utils.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o S4Vectors.so AEbufs.o DataFrame_class.o Hits_class.o LLint_class.o List_class.o R_init_S4Vectors.o Rle_class.o Rle_utils.o SEXP_utils.o SimpleList_class.o anyMissing.o character_utils.o eval_utils.o hash_utils.o integer_utils.o logical_utils.o map_ranges_to_runs.o raw_utils.o safe_arithm.o sort_utils.o subsetting_utils.o vector_utils.o -L/usr/lib/R/lib -lR\n","in method for ‘normalizeSingleBracketReplacementValue’ with signature ‘\"List\"’: no definition for class “List”\n","Creating a new generic function for ‘unname’ in package ‘S4Vectors’\n","Creating a new generic function for ‘expand.grid’ in package ‘S4Vectors’\n","Creating a new generic function for ‘findMatches’ in package ‘S4Vectors’\n","in method for ‘coerce’ with signature ‘\"Hits\",\"DFrame\"’: no definition for class “DFrame”\n","Creating a generic function for ‘as.factor’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘tabulate’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘cov’ from package ‘stats’ in package ‘S4Vectors’\n","Creating a generic function for ‘cor’ from package ‘stats’ in package ‘S4Vectors’\n","Creating a generic function for ‘smoothEnds’ from package ‘stats’ in package ‘S4Vectors’\n","Creating a generic function for ‘runmed’ from package ‘stats’ in package ‘S4Vectors’\n","Creating a generic function for ‘nchar’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘substr’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘substring’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘chartr’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘tolower’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘toupper’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘sub’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘gsub’ from package ‘base’ in package ‘S4Vectors’\n","Creating a generic function for ‘nlevels’ from package ‘base’ in package ‘S4Vectors’\n","in method for ‘coerce’ with signature ‘\"data.table\",\"DFrame\"’: no definition for class “data.table”\n","Creating a generic function for ‘complete.cases’ from package ‘stats’ in package ‘S4Vectors’\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Rinit.c -o Rinit.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c anyMissing.c -o anyMissing.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c envir.c -o envir.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c matchpt.c -o matchpt.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowMedians.c -o rowMedians.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c sublist_extract.c -o sublist_extract.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o Biobase.so Rinit.o anyMissing.o envir.o matchpt.o rowMedians.o sublist_extract.o -L/usr/lib/R/lib -lR\n","Creating a new generic function for ‘rowAlls’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colAlls’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowAnyNAs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colAnyNAs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowAnys’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colAnys’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowAvgsPerColSet’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colAvgsPerRowSet’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowCollapse’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colCollapse’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowCounts’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colCounts’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowCummaxs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colCummaxs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowCummins’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colCummins’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowCumprods’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colCumprods’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowCumsums’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colCumsums’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowIQRDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colIQRDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowIQRs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colIQRs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowLogSumExps’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colLogSumExps’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowMadDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colMadDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowMads’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colMads’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowMaxs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colMaxs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowMeans2’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colMeans2’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowMedians’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colMedians’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowMins’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colMins’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowOrderStats’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colOrderStats’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowProds’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colProds’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowQuantiles’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colQuantiles’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowRanges’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colRanges’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowRanks’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colRanks’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowSdDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colSdDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowSds’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colSds’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowSums2’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colSums2’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowTabulates’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colTabulates’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowVarDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colVarDiffs’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowVars’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colVars’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowWeightedMads’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colWeightedMads’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowWeightedMeans’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colWeightedMeans’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowWeightedMedians’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colWeightedMedians’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowWeightedSds’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colWeightedSds’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘rowWeightedVars’ in package ‘MatrixGenerics’\n","Creating a new generic function for ‘colWeightedVars’ in package ‘MatrixGenerics’\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c CompressedAtomicList_utils.c -o CompressedAtomicList_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c CompressedIRangesList_class.c -o CompressedIRangesList_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c CompressedList_class.c -o CompressedList_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Grouping_class.c -o Grouping_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c IPosRanges_comparison.c -o IPosRanges_comparison.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c IRanges_class.c -o IRanges_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c IRanges_constructor.c -o IRanges_constructor.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c NCList.c -o NCList.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_IRanges.c -o R_init_IRanges.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Ranges_class.c -o Ranges_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c RleViews_utils.c -o RleViews_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SimpleIRangesList_class.c -o SimpleIRangesList_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c coverage_methods.c -o coverage_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c extractListFragments.c -o extractListFragments.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c inter_range_methods.c -o inter_range_methods.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o IRanges.so CompressedAtomicList_utils.o CompressedIRangesList_class.o CompressedList_class.o Grouping_class.o IPosRanges_comparison.o IRanges_class.o IRanges_constructor.o NCList.o R_init_IRanges.o Ranges_class.o RleViews_utils.o S4Vectors_stubs.o SimpleIRangesList_class.o coverage_methods.o extractListFragments.o inter_range_methods.o -L/usr/lib/R/lib -lR\n","Creating a generic function for ‘drop’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘runmed’ from package ‘stats’ in package ‘IRanges’\n","Creating a generic function for ‘chartr’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘toupper’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘tolower’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘sub’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘gsub’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘startsWith’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘endsWith’ from package ‘base’ in package ‘IRanges’\n","Creating a generic function for ‘smoothEnds’ from package ‘stats’ in package ‘IRanges’\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c IRanges_stubs.c -o IRanges_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Ocopy_byteblocks.c -o Ocopy_byteblocks.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c RDS_random_access.c -o RDS_random_access.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_XVector.c -o R_init_XVector.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SharedDouble_class.c -o SharedDouble_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SharedInteger_class.c -o SharedInteger_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SharedRaw_class.c -o SharedRaw_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SharedVector_class.c -o SharedVector_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XDouble_class.c -o XDouble_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XInteger_class.c -o XInteger_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XRawList_comparison.c -o XRawList_comparison.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XRaw_class.c -o XRaw_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XVectorList_class.c -o XVectorList_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XVector_class.c -o XVector_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c io_utils.c -o io_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c slice_methods.c -o slice_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c vector_copy.c -o vector_copy.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c view_summarization_methods.c -o view_summarization_methods.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o XVector.so IRanges_stubs.o Ocopy_byteblocks.o RDS_random_access.o R_init_XVector.o S4Vectors_stubs.o SharedDouble_class.o SharedInteger_class.o SharedRaw_class.o SharedVector_class.o XDouble_class.o XInteger_class.o XRawList_comparison.o XRaw_class.o XVectorList_class.o XVector_class.o io_utils.o slice_methods.o vector_copy.o view_summarization_methods.o -lz -L/usr/lib/R/lib -lR\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_S4Arrays.c -o R_init_S4Arrays.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c abind.c -o abind.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c array_selection.c -o array_selection.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dim_tuning_utils.c -o dim_tuning_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowsum.c -o rowsum.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o S4Arrays.so R_init_S4Arrays.o S4Vectors_stubs.o abind.o array_selection.o dim_tuning_utils.o rowsum.o -L/usr/lib/R/lib -lR\n","Creating a new generic function for ‘rowsum’ in package ‘S4Arrays’\n","Creating a new generic function for ‘abind’ in package ‘S4Arrays’\n","checking whether the C++ compiler works... yes\n","checking for C++ compiler default output file name... a.out\n","checking for suffix of executables... \n","checking whether we are cross compiling... no\n","checking for suffix of object files... o\n","checking whether the compiler supports GNU C++... yes\n","checking whether g++ -std=gnu++17 accepts -g... yes\n","checking for g++ -std=gnu++17 option to enable C++11 features... none needed\n","checking for library containing shm_open... none required\n","checking for stdio.h... yes\n","checking for stdlib.h... yes\n","checking for string.h... yes\n","checking for inttypes.h... yes\n","checking for stdint.h... yes\n","checking for strings.h... yes\n","checking for sys/stat.h... yes\n","checking for sys/types.h... yes\n","checking for unistd.h... yes\n","checking for sys/mman.h... yes\n","configure: creating ./config.status\n","config.status: creating src/Makevars\n","g++ -std=gnu++11 -I\"/usr/share/R/include\" -DNDEBUG -I\"./\" -I'/usr/local/lib/R/site-library/BH/include' -I'/usr/lib/R/site-library/cpp11/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c cpp11.cpp -o cpp11.o\n","g++ -std=gnu++11 -I\"/usr/share/R/include\" -DNDEBUG -I\"./\" -I'/usr/local/lib/R/site-library/BH/include' -I'/usr/lib/R/site-library/cpp11/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c ipcmutex.cpp -o ipcmutex.o\n","g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o BiocParallel.so cpp11.o ipcmutex.o -L/usr/lib/R/lib -lR\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ExtendableJaggedArray.c -o ExtendableJaggedArray.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c IRanges_stubs.c -o IRanges_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c OPBufTree.c -o OPBufTree.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_SparseArray.c -o R_init_SparseArray.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Rvector_summarization.c -o Rvector_summarization.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c Rvector_utils.c -o Rvector_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SBT_utils.c -o SBT_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SVT_SparseArray_class.c -o SVT_SparseArray_class.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_Arith_methods.c -o SparseArray_Arith_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_Compare_methods.c -o SparseArray_Compare_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_Complex_methods.c -o SparseArray_Complex_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_Logic_methods.c -o SparseArray_Logic_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_Math_methods.c -o SparseArray_Math_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_abind.c -o SparseArray_abind.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_aperm.c -o SparseArray_aperm.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_dim_tuning.c -o SparseArray_dim_tuning.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_matrixStats.c -o SparseArray_matrixStats.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_misc_methods.c -o SparseArray_misc_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_subassignment.c -o SparseArray_subassignment.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_subassignment_OLD.c -o SparseArray_subassignment_OLD.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_subsetting.c -o SparseArray_subsetting.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseArray_summarization.c -o SparseArray_summarization.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseMatrix_mult.c -o SparseMatrix_mult.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseVec.c -o SparseVec.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseVec_Arith.c -o SparseVec_Arith.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseVec_Compare.c -o SparseVec_Compare.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseVec_Logic.c -o SparseVec_Logic.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseVec_Math.c -o SparseVec_Math.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c SparseVec_dotprod.c -o SparseVec_dotprod.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c XVector_stubs.c -o XVector_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c argcheck_utils.c -o argcheck_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c coerceVector2.c -o coerceVector2.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c leaf_utils.c -o leaf_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c randomSparseArray.c -o randomSparseArray.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c readSparseCSV.c -o readSparseCSV.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rowsum_methods.c -o rowsum_methods.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c sparseMatrix_utils.c -o sparseMatrix_utils.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c test.c -o test.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include' -I'/usr/local/lib/R/site-library/XVector/include'    -fopenmp -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c thread_control.c -o thread_control.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o SparseArray.so ExtendableJaggedArray.o IRanges_stubs.o OPBufTree.o R_init_SparseArray.o Rvector_summarization.o Rvector_utils.o S4Vectors_stubs.o SBT_utils.o SVT_SparseArray_class.o SparseArray_Arith_methods.o SparseArray_Compare_methods.o SparseArray_Complex_methods.o SparseArray_Logic_methods.o SparseArray_Math_methods.o SparseArray_abind.o SparseArray_aperm.o SparseArray_dim_tuning.o SparseArray_matrixStats.o SparseArray_misc_methods.o SparseArray_subassignment.o SparseArray_subassignment_OLD.o SparseArray_subsetting.o SparseArray_summarization.o SparseMatrix_mult.o SparseVec.o SparseVec_Arith.o SparseVec_Compare.o SparseVec_Logic.o SparseVec_Math.o SparseVec_dotprod.o XVector_stubs.o argcheck_utils.o coerceVector2.o leaf_utils.o randomSparseArray.o readSparseCSV.o rowsum_methods.o sparseMatrix_utils.o test.o thread_control.o -fopenmp -L/usr/lib/R/lib -lR\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c IRanges_stubs.c -o IRanges_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_GenomicRanges.c -o R_init_GenomicRanges.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include' -I'/usr/local/lib/R/site-library/IRanges/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c transcript_utils.c -o transcript_utils.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o GenomicRanges.so IRanges_stubs.o R_init_GenomicRanges.o S4Vectors_stubs.o transcript_utils.o -L/usr/lib/R/lib -lR\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c R_init_DelayedArray.c -o R_init_DelayedArray.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o\n","gcc -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/local/lib/R/site-library/S4Vectors/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c compress_atomic_vector.c -o compress_atomic_vector.o\n","gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o DelayedArray.so R_init_DelayedArray.o S4Vectors_stubs.o compress_atomic_vector.o -L/usr/lib/R/lib -lR\n","Creating a new generic function for ‘apply’ in package ‘DelayedArray’\n","Creating a new generic function for ‘sweep’ in package ‘DelayedArray’\n","Creating a new generic function for ‘scale’ in package ‘DelayedArray’\n","Creating a generic function for ‘dnorm’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘pnorm’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘qnorm’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘dbinom’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘pbinom’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘qbinom’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘dpois’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘ppois’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘qpois’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘dlogis’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘plogis’ from package ‘stats’ in package ‘DelayedArray’\n","Creating a generic function for ‘qlogis’ from package ‘stats’ in package ‘DelayedArray’\n","g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include' -I'/usr/local/lib/R/site-library/RcppArmadillo/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c DESeq2.cpp -o DESeq2.o\n","g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG  -I'/usr/lib/R/site-library/Rcpp/include' -I'/usr/local/lib/R/site-library/RcppArmadillo/include'     -fpic  -g -O2 -ffile-prefix-map=/build/r-base-0RQCNp/r-base-4.5.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c RcppExports.cpp -o RcppExports.o\n","g++ -std=gnu++17 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -o DESeq2.so DESeq2.o RcppExports.o -llapack -lblas -lgfortran -lm -lquadmath -L/usr/lib/R/lib -lR\n","> \n"]},{"output_type":"stream","name":"stderr","text":["Installing package into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","trying URL 'http://cran.us.r-project.org/src/contrib/BiocManager_1.30.26.tar.gz'\n","Content type 'application/x-gzip' length 594489 bytes (580 KB)\n","==================================================\n","downloaded 580 KB\n","\n","* installing *source* package ‘BiocManager’ ...\n","** this is package ‘BiocManager’ version ‘1.30.26’\n","** package ‘BiocManager’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (BiocManager)\n","\n","The downloaded source packages are in\n","\t‘/tmp/RtmpB1sVAP/downloaded_packages’\n","Bioconductor version 3.21 (BiocManager 1.30.26), R 4.5.1 (2025-06-13)\n","Installing package(s) 'BiocVersion', 'DESeq2'\n","also installing the dependencies ‘formatR’, ‘UCSC.utils’, ‘GenomeInfoDbData’, ‘abind’, ‘SparseArray’, ‘lambda.r’, ‘futile.options’, ‘GenomeInfoDb’, ‘XVector’, ‘S4Arrays’, ‘DelayedArray’, ‘futile.logger’, ‘snow’, ‘BH’, ‘S4Vectors’, ‘IRanges’, ‘GenomicRanges’, ‘SummarizedExperiment’, ‘BiocGenerics’, ‘Biobase’, ‘BiocParallel’, ‘matrixStats’, ‘locfit’, ‘MatrixGenerics’, ‘RcppArmadillo’\n","\n","trying URL 'https://cloud.r-project.org/src/contrib/formatR_1.14.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/UCSC.utils_1.4.0.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/data/annotation/src/contrib/GenomeInfoDbData_1.2.14.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/abind_1.4-8.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/SparseArray_1.8.1.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/lambda.r_1.2.4.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/futile.options_1.0.1.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/GenomeInfoDb_1.44.3.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/XVector_0.48.0.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/S4Arrays_1.8.1.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/DelayedArray_0.34.1.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/futile.logger_1.4.3.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/snow_0.4-4.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/BH_1.87.0-1.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/S4Vectors_0.46.0.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/IRanges_2.42.0.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/GenomicRanges_1.60.0.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/SummarizedExperiment_1.38.1.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/BiocGenerics_0.54.1.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/Biobase_2.68.0.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/BiocParallel_1.42.2.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/matrixStats_1.5.0.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/locfit_1.5-9.12.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/MatrixGenerics_1.20.0.tar.gz'\n","trying URL 'https://cloud.r-project.org/src/contrib/RcppArmadillo_15.0.2-2.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/BiocVersion_3.21.1.tar.gz'\n","trying URL 'https://bioconductor.org/packages/3.21/bioc/src/contrib/DESeq2_1.48.2.tar.gz'\n","* installing *source* package ‘formatR’ ...\n","** this is package ‘formatR’ version ‘1.14’\n","** package ‘formatR’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (formatR)\n","* installing *source* package ‘GenomeInfoDbData’ ...\n","** this is package ‘GenomeInfoDbData’ version ‘1.2.14’\n","** using staged installation\n","** data\n","** inst\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (GenomeInfoDbData)\n","* installing *source* package ‘abind’ ...\n","** this is package ‘abind’ version ‘1.4-8’\n","** package ‘abind’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (abind)\n","* installing *source* package ‘futile.options’ ...\n","** this is package ‘futile.options’ version ‘1.0.1’\n","** package ‘futile.options’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (futile.options)\n","* installing *source* package ‘snow’ ...\n","** this is package ‘snow’ version ‘0.4-4’\n","** package ‘snow’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (snow)\n","* installing *source* package ‘BH’ ...\n","** this is package ‘BH’ version ‘1.87.0-1’\n","** package ‘BH’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** inst\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (BH)\n","* installing *source* package ‘BiocGenerics’ ...\n","** this is package ‘BiocGenerics’ version ‘0.54.1’\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (BiocGenerics)\n","* installing *source* package ‘matrixStats’ ...\n","** this is package ‘matrixStats’ version ‘1.5.0’\n","** package ‘matrixStats’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-matrixStats/00new/matrixStats/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (matrixStats)\n","* installing *source* package ‘locfit’ ...\n","** this is package ‘locfit’ version ‘1.5-9.12’\n","** package ‘locfit’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-locfit/00new/locfit/libs\n","** R\n","** data\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (locfit)\n","* installing *source* package ‘RcppArmadillo’ ...\n","** this is package ‘RcppArmadillo’ version ‘15.0.2-2’\n","** package ‘RcppArmadillo’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** libs\n","using C++ compiler: ‘g++ (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-RcppArmadillo/00new/RcppArmadillo/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (RcppArmadillo)\n","* installing *source* package ‘BiocVersion’ ...\n","** this is package ‘BiocVersion’ version ‘3.21.1’\n","** using staged installation\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (BiocVersion)\n","* installing *source* package ‘lambda.r’ ...\n","** this is package ‘lambda.r’ version ‘1.2.4’\n","** package ‘lambda.r’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (lambda.r)\n","* installing *source* package ‘S4Vectors’ ...\n","** this is package ‘S4Vectors’ version ‘0.46.0’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","In file included from /usr/include/string.h:535,\n","                 from /usr/share/R/include/R_ext/RS.h:34,\n","                 from /usr/share/R/include/Rdefines.h:38,\n","                 from ../inst/include/S4Vectors_defines.h:18,\n","                 from S4Vectors.h:1,\n","                 from Hits_class.c:4:\n","In function ‘memcpy’,\n","    inlined from ‘qsort_hits’ at Hits_class.c:74:4:\n","/usr/include/x86_64-linux-gnu/bits/string_fortified.h:29:10: warning: ‘__builtin_memcpy’ specified bound between 18446744065119617024 and 18446744073709551612 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]\n","   29 |   return __builtin___memcpy_chk (__dest, __src, __len,\n","      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n","   30 |                                  __glibc_objsize0 (__dest));\n","      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~\n","installing to /usr/local/lib/R/site-library/00LOCK-S4Vectors/00new/S4Vectors/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (S4Vectors)\n","* installing *source* package ‘Biobase’ ...\n","** this is package ‘Biobase’ version ‘2.68.0’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-Biobase/00new/Biobase/libs\n","** R\n","** data\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (Biobase)\n","* installing *source* package ‘MatrixGenerics’ ...\n","** this is package ‘MatrixGenerics’ version ‘1.20.0’\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (MatrixGenerics)\n","* installing *source* package ‘UCSC.utils’ ...\n","** this is package ‘UCSC.utils’ version ‘1.4.0’\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (UCSC.utils)\n","* installing *source* package ‘futile.logger’ ...\n","** this is package ‘futile.logger’ version ‘1.4.3’\n","** package ‘futile.logger’ successfully unpacked and MD5 sums checked\n","** using staged installation\n","** R\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (futile.logger)\n","* installing *source* package ‘IRanges’ ...\n","** this is package ‘IRanges’ version ‘2.42.0’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-IRanges/00new/IRanges/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (IRanges)\n","* installing *source* package ‘GenomeInfoDb’ ...\n","** this is package ‘GenomeInfoDb’ version ‘1.44.3’\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (GenomeInfoDb)\n","* installing *source* package ‘XVector’ ...\n","** this is package ‘XVector’ version ‘0.48.0’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-XVector/00new/XVector/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (XVector)\n","* installing *source* package ‘S4Arrays’ ...\n","** this is package ‘S4Arrays’ version ‘1.8.1’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-S4Arrays/00new/S4Arrays/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (S4Arrays)\n","* installing *source* package ‘BiocParallel’ ...\n","** this is package ‘BiocParallel’ version ‘1.42.2’\n","** using staged installation\n","** libs\n","using C++ compiler: ‘g++ (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","using C++11\n","installing to /usr/local/lib/R/site-library/00LOCK-BiocParallel/00new/BiocParallel/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (BiocParallel)\n","* installing *source* package ‘SparseArray’ ...\n","** this is package ‘SparseArray’ version ‘1.8.1’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-SparseArray/00new/SparseArray/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (SparseArray)\n","* installing *source* package ‘GenomicRanges’ ...\n","** this is package ‘GenomicRanges’ version ‘1.60.0’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-GenomicRanges/00new/GenomicRanges/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (GenomicRanges)\n","* installing *source* package ‘DelayedArray’ ...\n","** this is package ‘DelayedArray’ version ‘0.34.1’\n","** using staged installation\n","** libs\n","using C compiler: ‘gcc (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","installing to /usr/local/lib/R/site-library/00LOCK-DelayedArray/00new/DelayedArray/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (DelayedArray)\n","* installing *source* package ‘SummarizedExperiment’ ...\n","** this is package ‘SummarizedExperiment’ version ‘1.38.1’\n","** using staged installation\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (SummarizedExperiment)\n","* installing *source* package ‘DESeq2’ ...\n","** this is package ‘DESeq2’ version ‘1.48.2’\n","** using staged installation\n","** libs\n","using C++ compiler: ‘g++ (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0’\n","DESeq2.cpp: In function ‘Rcpp::List fitDispGrid(SEXP, SEXP, SEXP, SEXP, SEXP, SEXP, SEXP, SEXP, SEXP, SEXP, SEXP)’:\n","DESeq2.cpp:501:19: warning: ‘eT arma::Mat<eT>::max(arma::uword&) const [with eT = double; arma::uword = unsigned int]’ is deprecated: use .index_max() instead [-Wdeprecated-declarations]\n","  501 |     logpostvec.max(idxmax);\n","      |     ~~~~~~~~~~~~~~^~~~~~~~\n","In file included from /usr/local/lib/R/site-library/RcppArmadillo/include/current/armadillo:201,\n","                 from /usr/local/lib/R/site-library/RcppArmadillo/include/RcppArmadillo/interface/RcppArmadilloForward.h:57,\n","                 from /usr/local/lib/R/site-library/RcppArmadillo/include/RcppArmadillo.h:32,\n","                 from DESeq2.cpp:16:\n","/usr/local/lib/R/site-library/RcppArmadillo/include/current/armadillo_bits/Mat_bones.hpp:545:56: note: declared here\n","  545 |   [[deprecated(\"use .index_max() instead\")]] inline eT max(uword& index_of_max_val) const;\n","      |                                                        ^~~\n","DESeq2.cpp:508:19: warning: ‘eT arma::Mat<eT>::max(arma::uword&) const [with eT = double; arma::uword = unsigned int]’ is deprecated: use .index_max() instead [-Wdeprecated-declarations]\n","  508 |     logpostvec.max(idxmax);\n","      |     ~~~~~~~~~~~~~~^~~~~~~~\n","In file included from /usr/local/lib/R/site-library/RcppArmadillo/include/current/armadillo:692,\n","                 from /usr/local/lib/R/site-library/RcppArmadillo/include/RcppArmadillo/interface/RcppArmadilloForward.h:57,\n","                 from /usr/local/lib/R/site-library/RcppArmadillo/include/RcppArmadillo.h:32,\n","                 from DESeq2.cpp:16:\n","/usr/local/lib/R/site-library/RcppArmadillo/include/current/armadillo_bits/Mat_meat.hpp:7850:1: note: declared here\n"," 7850 | Mat<eT>::max(uword& index_of_max_val) const\n","      | ^~~~~~~\n","installing to /usr/local/lib/R/site-library/00LOCK-DESeq2/00new/DESeq2/libs\n","** R\n","** inst\n","** byte-compile and prepare package for lazy loading\n","** help\n","*** installing help indices\n","** building package indices\n","** installing vignettes\n","** testing if installed package can be loaded from temporary location\n","** checking absolute paths in shared objects and dynamic libraries\n","** testing if installed package can be loaded from final location\n","** testing if installed package keeps a record of temporary installation path\n","* DONE (DESeq2)\n","\n","The downloaded source packages are in\n","\t‘/tmp/RtmpB1sVAP/downloaded_packages’\n"]}],"source":["%%script R --vanilla\n","install.packages(\"BiocManager\", repos=\"http://cran.us.r-project.org\")\n","BiocManager::install(\"DESeq2\", update=FALSE, ask=FALSE)"]},{"cell_type":"code","source":["!pip install -q rpy2\n","%load_ext rpy2.ipython"],"metadata":{"id":"ChG0RcfD9bN0"},"id":"ChG0RcfD9bN0","execution_count":null,"outputs":[]},{"cell_type":"markdown","source":["### 1.3) Upload raw RNAseq data - foldChange"],"metadata":{"id":"uoGAs0hc7srb"},"id":"uoGAs0hc7srb"},{"cell_type":"code","source":["from google.colab import drive\n","drive.mount('/content/drive')"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"dBeJFp_B_3L6","outputId":"65cac18d-e02c-49bd-dcf7-d983140100e7","executionInfo":{"status":"ok","timestamp":1761001736455,"user_tz":420,"elapsed":20827,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}}},"id":"dBeJFp_B_3L6","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}]},{"cell_type":"code","source":["    #Upload raw RNAseq data from google drive to feed to DESeq2\n","import pandas as pd\n","file_path = '/content/drive/My Drive/GenomicsProject_data/tcga_luad_raw_expression_data.csv'\n","counts_raw = pd.read_csv(file_path)"],"metadata":{"id":"-tqmomk6Lhl2"},"id":"-tqmomk6Lhl2","execution_count":null,"outputs":[]},{"cell_type":"code","source":["# Preprocess counts_raw in Python to set SampleID as index and transpose\n","counts_preprocessed = counts_raw.set_index('Unnamed: 0')\n","\n","# Preprocess meta to set SampleID as index\n","meta_processed = meta.set_index('SampleID')\n","\n","# Display the head of the preprocessed data to verify\n","print(\"Preprocessed counts preview (SampleIDs as index, genes as columns):\")\n","display(counts_preprocessed.head())\n","print(\"\\nPreprocessed metadata preview (SampleIDs as index):\")\n","display(meta_processed.head())"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":665},"id":"eR22Hy_GONaD","outputId":"4537e0b9-84fa-4ae3-fbf7-40ff65bcbd03","executionInfo":{"status":"ok","timestamp":1761001906133,"user_tz":420,"elapsed":95,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}}},"id":"eR22Hy_GONaD","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Preprocessed counts preview (SampleIDs as index, genes as columns):\n"]},{"output_type":"display_data","data":{"text/plain":["                 ARHGEF10L    HIF3A   RNF17    RNF10    RNF11    RNF13  \\\n","Unnamed: 0                                                               \n","TCGA-69-7978-01     9.9898   4.2598  0.4181  10.3657  11.1718  10.5897   \n","TCGA-62-8399-01    10.4257  11.6239  0.0000  11.5489  11.0200   9.2843   \n","TCGA-78-7539-01     9.6264   9.1362  1.1231  11.6692  10.4679  10.4649   \n","TCGA-50-5931-11     8.6835   9.4824  0.8221  11.7341  11.6787  11.5412   \n","TCGA-73-4658-01     9.2078   5.0288  0.0000  11.6209  11.3414  10.9376   \n","\n","                 GTF2IP1    REM1   MTVR2  RTN4RL2  ...   GNGT2   GNGT1  \\\n","Unnamed: 0                                         ...                   \n","TCGA-69-7978-01  12.2708  4.7670  0.0000   8.2023  ...  6.2348  0.0000   \n","TCGA-62-8399-01  12.1540  5.7125  0.4628   5.5819  ...  4.4464  1.3294   \n","TCGA-78-7539-01  12.6559  4.3943  0.3725   3.5365  ...  6.0400  3.9201   \n","TCGA-50-5931-11  11.9285  5.9466  0.8221   3.3528  ...  6.3782  0.0000   \n","TCGA-73-4658-01  12.0539  6.0942  0.0000   7.4156  ...  6.3898  1.1048   \n","\n","                  TULP3     PTRF   BCL6B    GSTK1    SELP     SELS  \\\n","Unnamed: 0                                                           \n","TCGA-69-7978-01  9.4520  12.7565  8.2668  11.2400  6.1209   9.8977   \n","TCGA-62-8399-01  9.5226  12.2100  8.5437  10.3491  8.6398   9.7315   \n","TCGA-78-7539-01  9.2765  10.6498  6.1814  11.1659  6.0970  10.3540   \n","TCGA-50-5931-11  8.6781  14.6956  9.7151  10.5910  9.5115  10.4914   \n","TCGA-73-4658-01  9.2697  13.0036  8.9786  10.6777  8.4187  10.3142   \n","\n","                      base_id  sample_type  \n","Unnamed: 0                                  \n","TCGA-69-7978-01  TCGA-69-7978        Tumor  \n","TCGA-62-8399-01  TCGA-62-8399        Tumor  \n","TCGA-78-7539-01  TCGA-78-7539        Tumor  \n","TCGA-50-5931-11  TCGA-50-5931       Normal  \n","TCGA-73-4658-01  TCGA-73-4658        Tumor  \n","\n","[5 rows x 20532 columns]"],"text/html":["\n","  <div id=\"df-a3627dfb-5468-45f0-9842-e48dcade63b8\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ARHGEF10L</th>\n","      <th>HIF3A</th>\n","      <th>RNF17</th>\n","      <th>RNF10</th>\n","      <th>RNF11</th>\n","      <th>RNF13</th>\n","      <th>GTF2IP1</th>\n","      <th>REM1</th>\n","      <th>MTVR2</th>\n","      <th>RTN4RL2</th>\n","      <th>...</th>\n","      <th>GNGT2</th>\n","      <th>GNGT1</th>\n","      <th>TULP3</th>\n","      <th>PTRF</th>\n","      <th>BCL6B</th>\n","      <th>GSTK1</th>\n","      <th>SELP</th>\n","      <th>SELS</th>\n","      <th>base_id</th>\n","      <th>sample_type</th>\n","    </tr>\n","    <tr>\n","      <th>Unnamed: 0</th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>TCGA-69-7978-01</th>\n","      <td>9.9898</td>\n","      <td>4.2598</td>\n","      <td>0.4181</td>\n","      <td>10.3657</td>\n","      <td>11.1718</td>\n","      <td>10.5897</td>\n","      <td>12.2708</td>\n","      <td>4.7670</td>\n","      <td>0.0000</td>\n","      <td>8.2023</td>\n","      <td>...</td>\n","      <td>6.2348</td>\n","      <td>0.0000</td>\n","      <td>9.4520</td>\n","      <td>12.7565</td>\n","      <td>8.2668</td>\n","      <td>11.2400</td>\n","      <td>6.1209</td>\n","      <td>9.8977</td>\n","      <td>TCGA-69-7978</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-62-8399-01</th>\n","      <td>10.4257</td>\n","      <td>11.6239</td>\n","      <td>0.0000</td>\n","      <td>11.5489</td>\n","      <td>11.0200</td>\n","      <td>9.2843</td>\n","      <td>12.1540</td>\n","      <td>5.7125</td>\n","      <td>0.4628</td>\n","      <td>5.5819</td>\n","      <td>...</td>\n","      <td>4.4464</td>\n","      <td>1.3294</td>\n","      <td>9.5226</td>\n","      <td>12.2100</td>\n","      <td>8.5437</td>\n","      <td>10.3491</td>\n","      <td>8.6398</td>\n","      <td>9.7315</td>\n","      <td>TCGA-62-8399</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-78-7539-01</th>\n","      <td>9.6264</td>\n","      <td>9.1362</td>\n","      <td>1.1231</td>\n","      <td>11.6692</td>\n","      <td>10.4679</td>\n","      <td>10.4649</td>\n","      <td>12.6559</td>\n","      <td>4.3943</td>\n","      <td>0.3725</td>\n","      <td>3.5365</td>\n","      <td>...</td>\n","      <td>6.0400</td>\n","      <td>3.9201</td>\n","      <td>9.2765</td>\n","      <td>10.6498</td>\n","      <td>6.1814</td>\n","      <td>11.1659</td>\n","      <td>6.0970</td>\n","      <td>10.3540</td>\n","      <td>TCGA-78-7539</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-50-5931-11</th>\n","      <td>8.6835</td>\n","      <td>9.4824</td>\n","      <td>0.8221</td>\n","      <td>11.7341</td>\n","      <td>11.6787</td>\n","      <td>11.5412</td>\n","      <td>11.9285</td>\n","      <td>5.9466</td>\n","      <td>0.8221</td>\n","      <td>3.3528</td>\n","      <td>...</td>\n","      <td>6.3782</td>\n","      <td>0.0000</td>\n","      <td>8.6781</td>\n","      <td>14.6956</td>\n","      <td>9.7151</td>\n","      <td>10.5910</td>\n","      <td>9.5115</td>\n","      <td>10.4914</td>\n","      <td>TCGA-50-5931</td>\n","      <td>Normal</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-73-4658-01</th>\n","      <td>9.2078</td>\n","      <td>5.0288</td>\n","      <td>0.0000</td>\n","      <td>11.6209</td>\n","      <td>11.3414</td>\n","      <td>10.9376</td>\n","      <td>12.0539</td>\n","      <td>6.0942</td>\n","      <td>0.0000</td>\n","      <td>7.4156</td>\n","      <td>...</td>\n","      <td>6.3898</td>\n","      <td>1.1048</td>\n","      <td>9.2697</td>\n","      <td>13.0036</td>\n","      <td>8.9786</td>\n","      <td>10.6777</td>\n","      <td>8.4187</td>\n","      <td>10.3142</td>\n","      <td>TCGA-73-4658</td>\n","      <td>Tumor</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>5 rows × 20532 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a3627dfb-5468-45f0-9842-e48dcade63b8')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-a3627dfb-5468-45f0-9842-e48dcade63b8 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-a3627dfb-5468-45f0-9842-e48dcade63b8');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-33fbaddb-6e68-41da-a9a8-5438c2658847\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-33fbaddb-6e68-41da-a9a8-5438c2658847')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-33fbaddb-6e68-41da-a9a8-5438c2658847 button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe"}},"metadata":{}},{"output_type":"stream","name":"stdout","text":["\n","Preprocessed metadata preview (SampleIDs as index):\n"]},{"output_type":"display_data","data":{"text/plain":["                condition     PatientID\n","SampleID                               \n","TCGA-44-6778-01     Tumor  TCGA-44-6778\n","TCGA-50-5931-01     Tumor  TCGA-50-5931\n","TCGA-44-6144-01     Tumor  TCGA-44-6144\n","TCGA-44-2668-01     Tumor  TCGA-44-2668\n","TCGA-44-2665-01     Tumor  TCGA-44-2665"],"text/html":["\n","  <div id=\"df-b42f5069-67f6-4de4-8862-a89e9a712544\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>condition</th>\n","      <th>PatientID</th>\n","    </tr>\n","    <tr>\n","      <th>SampleID</th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>TCGA-44-6778-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-6778</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-50-5931-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-50-5931</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-6144-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-6144</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-2668-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-2668</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-2665-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-2665</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-b42f5069-67f6-4de4-8862-a89e9a712544')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-b42f5069-67f6-4de4-8862-a89e9a712544 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-b42f5069-67f6-4de4-8862-a89e9a712544');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-62d6cc15-a29f-4870-a3b7-9d89b54c900c\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-62d6cc15-a29f-4870-a3b7-9d89b54c900c')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-62d6cc15-a29f-4870-a3b7-9d89b54c900c button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","summary":"{\n  \"name\": \"display(meta_processed\",\n  \"rows\": 5,\n  \"fields\": [\n    {\n      \"column\": \"SampleID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"TCGA-50-5931-01\",\n          \"TCGA-44-2665-01\",\n          \"TCGA-44-6144-01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"condition\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"Tumor\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PatientID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"TCGA-50-5931\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"}},"metadata":{}}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":701},"id":"afccc7d2","outputId":"d62e9204-59b2-44a1-ead6-bd2cd5631114","executionInfo":{"status":"ok","timestamp":1761002046161,"user_tz":420,"elapsed":95,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}}},"source":["# Filter the raw RNA-seq data file for tumor/normal samples with matching methylation and expression data\n","# valid_paired DataFrame has 'SampleID_Tumor' and 'SampleID_Normal' columns\n","valid_sample_ids = list(valid_paired['SampleID_Tumor']) + list(valid_paired['SampleID_Normal'])\n","\n","# Filter counts_preprocessed and meta_processed to keep only the valid paired sample IDs\n","counts_filtered = counts_preprocessed.loc[valid_sample_ids]\n","meta_filtered = meta_processed.loc[valid_sample_ids]\n","\n","# Display the head of the filtered data to verify\n","print(\"Filtered counts preview (SampleIDs as index, genes as columns):\")\n","display(counts_filtered.head())\n","print(\"\\nFiltered metadata preview (SampleIDs as index):\")\n","display(meta_filtered.head())\n","print(\"\\nNumber of samples after filtering:\", counts_filtered.shape[0])\n"],"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Filtered counts preview (SampleIDs as index, genes as columns):\n"]},{"output_type":"display_data","data":{"text/plain":["                 ARHGEF10L   HIF3A   RNF17    RNF10    RNF11    RNF13  \\\n","Unnamed: 0                                                              \n","TCGA-44-6778-01     9.6205  7.9642  1.5378  11.8432  11.0531  10.9005   \n","TCGA-50-5931-01     9.4932  7.6474  0.7060  12.0061  10.9403  10.2340   \n","TCGA-44-2668-01     9.4032  3.8726  0.0000  11.4589  11.1915  10.8651   \n","TCGA-44-2665-01     9.5904  6.6864  0.9626  11.5975  10.9343  10.9350   \n","TCGA-44-6148-01     8.9758  8.8888  0.6189  11.7523  11.0227  10.6612   \n","\n","                 GTF2IP1    REM1   MTVR2  RTN4RL2  ...   GNGT2   GNGT1  \\\n","Unnamed: 0                                         ...                   \n","TCGA-44-6778-01  12.4145  4.5366  2.0609   4.1805  ...  6.1839  4.1291   \n","TCGA-50-5931-01  12.5008  3.1003  1.1780   9.6736  ...  5.3270  1.8177   \n","TCGA-44-2668-01  11.3604  3.8454  0.0000   5.5501  ...  5.0415  0.0000   \n","TCGA-44-2665-01  12.0246  3.7370  0.0000   8.7106  ...  4.7343  0.9626   \n","TCGA-44-6148-01  12.5022  5.1410  0.0000   6.6141  ...  4.3037  0.0000   \n","\n","                  TULP3     PTRF   BCL6B    GSTK1    SELP     SELS  \\\n","Unnamed: 0                                                           \n","TCGA-44-6778-01  8.9832  12.3412  9.0862  10.4779  9.4517  10.4395   \n","TCGA-50-5931-01  9.3925  11.2292  7.0270   9.7507  6.0857   9.6511   \n","TCGA-44-2668-01  9.8020  12.6297  8.0584  10.5011  6.5682  10.4597   \n","TCGA-44-2665-01  9.9923  12.8440  8.1411  10.9781  4.2499  10.4428   \n","TCGA-44-6148-01  9.6568  12.2590  7.1915  10.8171  7.5584   9.9243   \n","\n","                      base_id  sample_type  \n","Unnamed: 0                                  \n","TCGA-44-6778-01  TCGA-44-6778        Tumor  \n","TCGA-50-5931-01  TCGA-50-5931        Tumor  \n","TCGA-44-2668-01  TCGA-44-2668        Tumor  \n","TCGA-44-2665-01  TCGA-44-2665        Tumor  \n","TCGA-44-6148-01  TCGA-44-6148        Tumor  \n","\n","[5 rows x 20532 columns]"],"text/html":["\n","  <div id=\"df-68510024-61f3-4120-8f87-bd1915855f89\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ARHGEF10L</th>\n","      <th>HIF3A</th>\n","      <th>RNF17</th>\n","      <th>RNF10</th>\n","      <th>RNF11</th>\n","      <th>RNF13</th>\n","      <th>GTF2IP1</th>\n","      <th>REM1</th>\n","      <th>MTVR2</th>\n","      <th>RTN4RL2</th>\n","      <th>...</th>\n","      <th>GNGT2</th>\n","      <th>GNGT1</th>\n","      <th>TULP3</th>\n","      <th>PTRF</th>\n","      <th>BCL6B</th>\n","      <th>GSTK1</th>\n","      <th>SELP</th>\n","      <th>SELS</th>\n","      <th>base_id</th>\n","      <th>sample_type</th>\n","    </tr>\n","    <tr>\n","      <th>Unnamed: 0</th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>TCGA-44-6778-01</th>\n","      <td>9.6205</td>\n","      <td>7.9642</td>\n","      <td>1.5378</td>\n","      <td>11.8432</td>\n","      <td>11.0531</td>\n","      <td>10.9005</td>\n","      <td>12.4145</td>\n","      <td>4.5366</td>\n","      <td>2.0609</td>\n","      <td>4.1805</td>\n","      <td>...</td>\n","      <td>6.1839</td>\n","      <td>4.1291</td>\n","      <td>8.9832</td>\n","      <td>12.3412</td>\n","      <td>9.0862</td>\n","      <td>10.4779</td>\n","      <td>9.4517</td>\n","      <td>10.4395</td>\n","      <td>TCGA-44-6778</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-50-5931-01</th>\n","      <td>9.4932</td>\n","      <td>7.6474</td>\n","      <td>0.7060</td>\n","      <td>12.0061</td>\n","      <td>10.9403</td>\n","      <td>10.2340</td>\n","      <td>12.5008</td>\n","      <td>3.1003</td>\n","      <td>1.1780</td>\n","      <td>9.6736</td>\n","      <td>...</td>\n","      <td>5.3270</td>\n","      <td>1.8177</td>\n","      <td>9.3925</td>\n","      <td>11.2292</td>\n","      <td>7.0270</td>\n","      <td>9.7507</td>\n","      <td>6.0857</td>\n","      <td>9.6511</td>\n","      <td>TCGA-50-5931</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-2668-01</th>\n","      <td>9.4032</td>\n","      <td>3.8726</td>\n","      <td>0.0000</td>\n","      <td>11.4589</td>\n","      <td>11.1915</td>\n","      <td>10.8651</td>\n","      <td>11.3604</td>\n","      <td>3.8454</td>\n","      <td>0.0000</td>\n","      <td>5.5501</td>\n","      <td>...</td>\n","      <td>5.0415</td>\n","      <td>0.0000</td>\n","      <td>9.8020</td>\n","      <td>12.6297</td>\n","      <td>8.0584</td>\n","      <td>10.5011</td>\n","      <td>6.5682</td>\n","      <td>10.4597</td>\n","      <td>TCGA-44-2668</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-2665-01</th>\n","      <td>9.5904</td>\n","      <td>6.6864</td>\n","      <td>0.9626</td>\n","      <td>11.5975</td>\n","      <td>10.9343</td>\n","      <td>10.9350</td>\n","      <td>12.0246</td>\n","      <td>3.7370</td>\n","      <td>0.0000</td>\n","      <td>8.7106</td>\n","      <td>...</td>\n","      <td>4.7343</td>\n","      <td>0.9626</td>\n","      <td>9.9923</td>\n","      <td>12.8440</td>\n","      <td>8.1411</td>\n","      <td>10.9781</td>\n","      <td>4.2499</td>\n","      <td>10.4428</td>\n","      <td>TCGA-44-2665</td>\n","      <td>Tumor</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-6148-01</th>\n","      <td>8.9758</td>\n","      <td>8.8888</td>\n","      <td>0.6189</td>\n","      <td>11.7523</td>\n","      <td>11.0227</td>\n","      <td>10.6612</td>\n","      <td>12.5022</td>\n","      <td>5.1410</td>\n","      <td>0.0000</td>\n","      <td>6.6141</td>\n","      <td>...</td>\n","      <td>4.3037</td>\n","      <td>0.0000</td>\n","      <td>9.6568</td>\n","      <td>12.2590</td>\n","      <td>7.1915</td>\n","      <td>10.8171</td>\n","      <td>7.5584</td>\n","      <td>9.9243</td>\n","      <td>TCGA-44-6148</td>\n","      <td>Tumor</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>5 rows × 20532 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-68510024-61f3-4120-8f87-bd1915855f89')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-68510024-61f3-4120-8f87-bd1915855f89 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-68510024-61f3-4120-8f87-bd1915855f89');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-30b448f0-70f3-4c74-8cfd-7afeeb4d41c0\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-30b448f0-70f3-4c74-8cfd-7afeeb4d41c0')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-30b448f0-70f3-4c74-8cfd-7afeeb4d41c0 button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe"}},"metadata":{}},{"output_type":"stream","name":"stdout","text":["\n","Filtered metadata preview (SampleIDs as index):\n"]},{"output_type":"display_data","data":{"text/plain":["                condition     PatientID\n","SampleID                               \n","TCGA-44-6778-01     Tumor  TCGA-44-6778\n","TCGA-50-5931-01     Tumor  TCGA-50-5931\n","TCGA-44-2668-01     Tumor  TCGA-44-2668\n","TCGA-44-2665-01     Tumor  TCGA-44-2665\n","TCGA-44-6148-01     Tumor  TCGA-44-6148"],"text/html":["\n","  <div id=\"df-ae5b3018-d0cc-404d-b0a2-9f917edcb58c\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>condition</th>\n","      <th>PatientID</th>\n","    </tr>\n","    <tr>\n","      <th>SampleID</th>\n","      <th></th>\n","      <th></th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>TCGA-44-6778-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-6778</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-50-5931-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-50-5931</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-2668-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-2668</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-2665-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-2665</td>\n","    </tr>\n","    <tr>\n","      <th>TCGA-44-6148-01</th>\n","      <td>Tumor</td>\n","      <td>TCGA-44-6148</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-ae5b3018-d0cc-404d-b0a2-9f917edcb58c')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-ae5b3018-d0cc-404d-b0a2-9f917edcb58c button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-ae5b3018-d0cc-404d-b0a2-9f917edcb58c');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-5f7bc745-de5b-4231-bbd0-2b8569db8ab4\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-5f7bc745-de5b-4231-bbd0-2b8569db8ab4')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-5f7bc745-de5b-4231-bbd0-2b8569db8ab4 button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","summary":"{\n  \"name\": \"print(\\\"\\\\nNumber of samples after filtering:\\\", counts_filtered\",\n  \"rows\": 5,\n  \"fields\": [\n    {\n      \"column\": \"SampleID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"TCGA-50-5931-01\",\n          \"TCGA-44-6148-01\",\n          \"TCGA-44-2668-01\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"condition\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"Tumor\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PatientID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"TCGA-50-5931\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"}},"metadata":{}},{"output_type":"stream","name":"stdout","text":["\n","Number of samples after filtering: 36\n"]}],"id":"afccc7d2"},{"cell_type":"markdown","source":["## 2) RNAseq analysis using DESeq2 library"],"metadata":{"id":"5mZQit2pBY2Y"},"id":"5mZQit2pBY2Y"},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":734},"outputId":"e1375a84-5be0-43a0-c3b7-50926dc4f2f7","executionInfo":{"status":"ok","timestamp":1761002799767,"user_tz":420,"elapsed":52405,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}},"collapsed":true,"id":"HFBH5_PU8QqD"},"outputs":[{"output_type":"stream","name":"stdout","text":["Sample IDs from counts columns (after filtering): TCGA-44-6778-01 TCGA-50-5931-01 TCGA-44-2668-01 TCGA-44-2665-01 TCGA-44-6148-01 TCGA-50-5933-01 TCGA-44-6146-01 TCGA-50-5936-01 TCGA-49-6745-01 TCGA-50-5932-01 TCGA-44-6145-01 TCGA-44-6147-01 TCGA-50-5939-01 TCGA-44-5645-01 TCGA-38-4632-01 TCGA-50-5935-01 TCGA-50-5930-01 TCGA-73-4676-01 TCGA-44-6778-11 TCGA-50-5931-11 TCGA-44-2668-11 TCGA-44-2665-11 TCGA-44-6148-11 TCGA-50-5933-11 TCGA-44-6146-11 TCGA-50-5936-11 TCGA-49-6745-11 TCGA-50-5932-11 TCGA-44-6145-11 TCGA-44-6147-11 TCGA-50-5939-11 TCGA-44-5645-11 TCGA-38-4632-11 TCGA-50-5935-11 TCGA-50-5930-11 TCGA-73-4676-11 \n","Sample IDs from meta row names (after filtering): TCGA-44-6778-01 TCGA-50-5931-01 TCGA-44-2668-01 TCGA-44-2665-01 TCGA-44-6148-01 TCGA-50-5933-01 TCGA-44-6146-01 TCGA-50-5936-01 TCGA-49-6745-01 TCGA-50-5932-01 TCGA-44-6145-01 TCGA-44-6147-01 TCGA-50-5939-01 TCGA-44-5645-01 TCGA-38-4632-01 TCGA-50-5935-01 TCGA-50-5930-01 TCGA-73-4676-01 TCGA-44-6778-11 TCGA-50-5931-11 TCGA-44-2668-11 TCGA-44-2665-11 TCGA-44-6148-11 TCGA-50-5933-11 TCGA-44-6146-11 TCGA-50-5936-11 TCGA-49-6745-11 TCGA-50-5932-11 TCGA-44-6145-11 TCGA-44-6147-11 TCGA-50-5939-11 TCGA-44-5645-11 TCGA-38-4632-11 TCGA-50-5935-11 TCGA-50-5930-11 TCGA-73-4676-11 \n","Number of sample IDs in counts (after filtering): 36 \n","Number of sample IDs in meta (after filtering): 36 \n","DESeq2 analysis complete."]},{"output_type":"display_data","data":{"text/plain":["Installing package into ‘/usr/local/lib/R/site-library’\n","(as ‘lib’ is unspecified)\n","trying URL 'http://cran.us.r-project.org/src/contrib/ashr_2.2-63.tar.gz'\n","Content type 'application/x-gzip' length 935117 bytes (913 KB)\n","==================================================\n","downloaded 913 KB\n","\n","\n","The downloaded source packages are in\n","\t‘/tmp/RtmpDUrCR5/downloaded_packages’\n","  Note: levels of factors in the design contain characters other than\n","  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n","  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n","  for column names in R. [This is a message, not a warning or an error]\n","estimating size factors\n","  Note: levels of factors in the design contain characters other than\n","  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n","  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n","  for column names in R. [This is a message, not a warning or an error]\n","estimating dispersions\n","gene-wise dispersion estimates\n","mean-dispersion relationship\n","-- note: fitType='parametric', but the dispersion trend was not well captured by the\n","   function: y = a/x + b, and a local regression fit was automatically substituted.\n","   specify fitType='local' or 'mean' to avoid this message next time.\n","  Note: levels of factors in the design contain characters other than\n","  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n","  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n","  for column names in R. [This is a message, not a warning or an error]\n","final dispersion estimates\n","fitting model and testing\n","using 'ashr' for LFC shrinkage. If used in published research, please cite:\n","    Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.\n","    https://doi.org/10.1093/biostatistics/kxw041\n"]},"metadata":{}}],"source":["%%R -i counts_filtered -i meta_filtered -o res_df_r\n","library(DESeq2)\n","install.packages(\"ashr\", repos=\"http://cran.us.r-project.org\") # Install ashr package\n","library(ashr) # Load ashr library\n","\n","\n","# counts_filtered (SampleIDs as index, genes as columns) and\n","# meta_filtered (SampleIDs as index) dataframes are now available in R.\n","\n","# Remove non-gene columns from counts_filtered before transposing\n","counts_filtered_genes <- counts_filtered[, !(colnames(counts_filtered) %in% c(\"base_id\", \"sample_type\"))]\n","\n","\n","# Transpose the counts data to get genes as rows and samples as columns,\n","# which is the required format for DESeq2.\n","counts_raw_deseq2 <- t(as.matrix(counts_filtered_genes))\n","\n","\n","# Convert counts to a numeric matrix and ensure it's integer type\n","# Coerce errors to NA and then replace NAs with 0, as count data must be non-negative\n","counts_raw_deseq2 <- apply(counts_raw_deseq2, 2, function(x) as.numeric(as.character(x))) # Ensure numeric conversion\n","counts_raw_deseq2[is.na(counts_raw_deseq2)] <- 0\n","storage.mode(counts_raw_deseq2) <- \"integer\"\n","rownames(counts_raw_deseq2) <- colnames(counts_filtered_genes)\n","\n","# Ensure metadata has SampleID as row names in R and is in the same order as counts columns\n","meta <- meta_filtered\n","\n","# Ensure Condition and PatientID are factors with correct levels\n","meta$condition <- factor(meta$condition, levels=c(\"Normal\", \"Tumor\"))\n","meta$PatientID <- factor(meta$PatientID)\n","\n","# Remove genes with all zero counts after transposition and type conversion\n","counts_raw_deseq2 <- counts_raw_deseq2[rowSums(counts_raw_deseq2) > 0, ]\n","\n","# Ensure metadata only contains samples present in the counts matrix after gene filtering\n","meta <- meta[colnames(counts_raw_deseq2), , drop = FALSE]\n","\n","dds <- DESeqDataSetFromMatrix(countData=counts_raw_deseq2, colData=meta, design=~ PatientID + condition)\n","# Optional: filter out genes with low counts (e.g., sum of counts across all samples <= 10)\n","# Filtering can be done before creating the dds object or after\n","# dds <- dds[rowSums(counts(dds)) > 10, ]\n","\n","dds <- DESeq(dds)\n","\n","res <- results(dds, contrast=c(\"condition\", \"Tumor\", \"Normal\"))\n","res <- lfcShrink(dds, coef=\"condition_Tumor_vs_Normal\", type=\"ashr\")\n","\n","res_df_r <- as.data.frame(res)\n","rownames(res_df_r) <- rownames(res) # Explicitly set row names of the data frame to rownames of the results object\n","# The res_df_r dataframe will be passed back to the Python kernel\n","cat(\"DESeq2 analysis complete.\")"],"id":"HFBH5_PU8QqD"},{"cell_type":"markdown","source":[],"metadata":{"id":"JsrLRi_9CC5E"},"id":"JsrLRi_9CC5E"},{"cell_type":"markdown","source":["## 3) Saving the top DEGs using fdr and fold_change values"],"metadata":{"id":"WNwij4HfCEB_"},"id":"WNwij4HfCEB_"},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ac461b5b","outputId":"69df9f9e-52b5-468d-b331-135949349381","executionInfo":{"status":"ok","timestamp":1761003644972,"user_tz":420,"elapsed":22,"user":{"displayName":"sahar nouhzade malekshah","userId":"10460245873262824779"}}},"source":["# Define significance thresholds (same as used before)\n","fdr_threshold = 0.05\n","log2fc_threshold = 1\n","\n","# Filter the DESeq2 results DataFrame to get the DEGs\n","deseq2_degs_df = res_df_r[\n","    (res_df_r['padj'] < fdr_threshold) & (abs(res_df_r['log2FoldChange']) > log2fc_threshold)\n","]\n","\n","# Save the filtered DataFrame to a CSV file\n","# Keep the index (gene names) as a column in the CSV\n","deseq2_degs_df.to_csv(\"DESeq2_paired_DEG_results.csv\", index=True)\n","\n","print(\"DESeq2 differentially expressed genes saved to DESeq2_paired_DEG_results.csv\")"],"id":"ac461b5b","execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["DESeq2 differentially expressed genes saved to DESeq2_paired_DEG_results.csv\n"]}]}],"metadata":{"colab":{"provenance":[{"file_id":"1SDV03lCyYtni3S27CjliFUSxjnt6jsoV","timestamp":1760987916003}],"machine_shape":"hm"},"language_info":{"name":"python"},"kernelspec":{"name":"python3","display_name":"Python 3"}},"nbformat":4,"nbformat_minor":5}